src/main/java/com/powsybl/openloadflow/sensi/AbstractDcSensitivityAnalysis.java

```diff
@@ -0,0 +1,303 @@
+        DcLoadFlowResult dcLoadFlowResult = new DcLoadFlowEngine(lfNetworks, dcLoadFlowParameters).run();
+        for (LfBranch branch : dcLoadFlowResult.getNetwork().getBranches()) {
+            functionReferenceByBranch.put(branch.getId(), branch.getP1() * PerUnit.SB);
+        }
+        return functionReferenceByBranch;
+    }
+
+    protected void fillRhsSensitivityVariable(LfNetwork lfNetwork, EquationSystem equationSystem, List<SensitivityFactorGroup> factorGroups, Matrix rhs) {
+        for (SensitivityFactorGroup factorGroup : factorGroups) {
+            factorGroup.fillRhs(lfNetwork, equationSystem, rhs);
+        }
+    }
+
+    protected List<SensitivityFactorGroup> createFactorGroups(Network network, List<SensitivityFactor> factors) {
+        Map<String, SensitivityFactorGroup> groupIndexedById = new HashMap<>(factors.size());
+        // index factors by variable config
+        for (SensitivityFactor<?, ?> factor : factors) {
+            if (factor instanceof BranchFlowPerInjectionIncrease) {
+                BranchFlowPerInjectionIncrease injectionFactor = (BranchFlowPerInjectionIncrease) factor;
+                Injection<?> injection = getInjection(network, injectionFactor.getVariable().getInjectionId());
+                Bus bus = injection.getTerminal().getBusView().getBus();
+                // skip disconnected injections
+                if (bus != null) {
+                    groupIndexedById.computeIfAbsent(bus.getId(), id -> new InjectionFactorGroup(bus.getId())).addFactor(injectionFactor);
+                }
+            } else if (factor instanceof BranchFlowPerPSTAngle) {
+                BranchFlowPerPSTAngle pstAngleFactor = (BranchFlowPerPSTAngle) factor;
+                String phaseTapChangerHolderId = pstAngleFactor.getVariable().getPhaseTapChangerHolderId();
+                TwoWindingsTransformer twt = network.getTwoWindingsTransformer(phaseTapChangerHolderId);
+                if (twt == null) {
+                    throw new PowsyblException("Phase shifter '" + phaseTapChangerHolderId + "' not found");
+                }
+                groupIndexedById.computeIfAbsent(phaseTapChangerHolderId, k -> new PhaseTapFactorGroup(phaseTapChangerHolderId)).addFactor(pstAngleFactor);
+            } else {
+                throw new UnsupportedOperationException("Factor type '" + factor.getClass().getSimpleName() + "' not yet supported");
+            }
+        }
+
+        // assign an index to each factor group
+        int index = 0;
+        for (SensitivityFactorGroup factorGroup : groupIndexedById.values()) {
+            factorGroup.setIndex(index++);
+        }
+
+        return new ArrayList<>(groupIndexedById.values());
+    }
+
+    protected void computeFactorsInjection(Function<String, Map<String, Double>> getParticipationForBus, List<SensitivityFactorGroup> factorsGroups, Map<SensitivityFactor, Double> predefinedResult) {
```

Comment:
Not clear to me what is the purpose of this method?

Reply:
Each FactorGroup will be represented by a column in the rhs. This method is used to know what data will be put in this column (this data changes depending on the elements that will be participating in the slack distribution, so it needs to be called for every loss of connectivity).